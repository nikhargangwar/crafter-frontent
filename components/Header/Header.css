@import url('https://fonts.googleapis.com/css2?family=Big+Shoulders+Inline+Display:wght@500;600;700&family=Big+Shoulders+Text:wght@700&family=Sofia+Sans:wght@700&display=swap');

.header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #211E1E;
  padding: 10px 30px;
  /* border-bottom: 1px solid #18CCCC; */
  height: 10%;
  width: 100vw;
}

.logo {
  margin-left: 80px;
  margin-top: 10px;
  width: 50px;
}

.user {
  display: flex;
  align-items: center;
  justify-content: space-around;
}

.profile {
  display: flex;
  align-items: flex-start;
  flex-direction: column;
  justify-content: center;
}

.profile p {
  width: 150px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

p {
  margin: 0;
}

.user-name {
  font-size: 2vh;
  text-transform: capitalize;
  /* color: white; */
  font-family: 'Big Shoulders Inline Display', cursive;
  font-family: 'Big Shoulders Text', cursive;
  font-family: 'Sofia Sans', sans-serif;
}

.avatar {
  font-size: 3vh !important;
  text-transform: uppercase;
}

.profile-menu {
  background-color: #ecf1f1;
  padding: 18px 15px;
  margin: 0 10px;
  border-radius: 10px;
}

.logout {
  height: 20% !important;
  margin: 0 10px;
}