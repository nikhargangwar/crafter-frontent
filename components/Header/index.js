import React from 'react';
import './Header.css';
import { Avatar, IconButton, Menu, MenuItem,Divider, ListItemIcon, Box } from '@mui/material';
import { useNavigate } from 'react-router';
import logo from '../../assets/crafter-logo-dark.svg';
import { Logout, OpenInNew } from '@mui/icons-material';
import AssignmentTurnedInOutlinedIcon from '@mui/icons-material/AssignmentTurnedInOutlined';
import ToggleColorMode from '../ToggleColorMode';

const Header = () => {
  const [anchorEl, setAnchorEl] = React.useState(null);
  const navigate = useNavigate();

  const open = Boolean(anchorEl);
  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };
  const handleClose = () => {
    setAnchorEl(null);
  };
  const userDetails = JSON.parse(localStorage.getItem('user_detail'));
  const handleLogout = () => {
    localStorage.clear();
    navigate('/login');
  };
  const handleTasks = () => {
    navigate('/tasks');
  };
  return (
    <>
      <Box className='flex items-center justify-between p-3 h-[10%] w-full'>
        <div className='flex items-center w-14 ml-14 cursor-pointer' onClick={() => navigate('/home')}>
          <img src={logo}/>
        </div>
        <div className='user'>
          <ToggleColorMode/>
          <IconButton
            onClick={handleClick}
            size="small"
            sx={{ ml: 2 }}
            aria-controls={open ? 'account-menu' : undefined}
            aria-haspopup="true"
            aria-expanded={open ? 'true' : undefined}
          >
            <Avatar sx={{ width: 45, height: 45, bgcolor: '#A7CCCC', color: 'black' }} className='avatar'>{userDetails?.fullName.slice(0,1)}</Avatar>
          </IconButton>
        </div>
      </Box>
      <Menu
        className='menu'
        anchorEl={anchorEl}
        id="account-menu"
        open={open}
        onClose={handleClose}
        onClick={handleClose}
        PaperProps={{
          elevation: 0,
          sx: {
            overflow: 'visible',
            width: 250,
            height: 210,
            borderRadius: 4,
            filter: 'drop-shadow(0px 3px 8px rgba(0,0,0,0.32))',
            mt: 1.5,
            '& .MuiAvatar-root': {
              width: 60,
              height: 60,
              ml: -0.5,
              mr: 1,
            },
            '&:before': {
              content: '""',
              display: 'block',
              position: 'absolute',
              top: 0,
              right: 23,
              width: 10,
              height: 10,
              bgcolor: 'background.paper',
              transform: 'translateY(-50%) rotate(45deg)',
              zIndex: 0,
            },
          },
        }}
        transformOrigin={{ horizontal: 'right', vertical: 'top' }}
        anchorOrigin={{ horizontal: 'right', vertical: 'bottom' }}
      >
        <MenuItem className='profile-menu' onClick={handleClose}>
          <Avatar sx={{ width: 90, height: 90 , bgcolor: '#A7CCCC', color: 'black'}}  className='avatar'>{userDetails?.fullName.slice(0,1)}</Avatar>
          <div className='profile'>
            <p className='user-name'>{userDetails?.fullName}</p>
            <p>{userDetails?.email}</p>
          </div>
        </MenuItem>
        <Divider />
        <a href='https://k8s-designer.onrender.com/' target='__blank'>
          <MenuItem className='logout'>
            <ListItemIcon>
              <OpenInNew fontSize="small" />
            </ListItemIcon>
          Documentation
          </MenuItem>
        </a>
        <MenuItem className='tasks' onClick={handleTasks}>
          <ListItemIcon>
            <AssignmentTurnedInOutlinedIcon fontSize="small" />
          </ListItemIcon>
          Tasks
        </MenuItem>
        <MenuItem className='logout' onClick={handleLogout}>
          <ListItemIcon>
            <Logout fontSize="small" />
          </ListItemIcon>
          Logout
        </MenuItem>
      </Menu>
    </>
  );
};

export default Header;

