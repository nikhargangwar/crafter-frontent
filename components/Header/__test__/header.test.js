import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '..';

describe('Home Page', () => {
  it('should render correctly', () => {
    jest.spyOn(Storage.prototype, 'getItem').mockImplementation((key) => {
      switch (key) {
      case 'user_detail':
        return JSON.stringify({ email: 'abc@gmail.com', fullName: 'name name' });
      default:
        return null;
      }
    });
    const { asFragment } = render(<Router><Header /></Router>);
    expect(asFragment()).toMatchSnapshot();
  });
});
