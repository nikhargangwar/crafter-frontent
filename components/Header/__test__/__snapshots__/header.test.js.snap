// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page should render correctly 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-between p-3 h-[10%] w-full MuiBox-root css-0"
  >
    <div
      class="flex items-center w-14 ml-14 cursor-pointer"
    >
      <img
        src="crafter-logo-dark.svg"
      />
    </div>
    <div
      class="user"
    >
      <div
        class="MuiBox-root css-j0tqbx"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-whjmaz-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="Brightness4Icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <button
        aria-haspopup="true"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-137ul40-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-1d090hw-MuiAvatar-root"
        >
          n
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
